### PR TITLE
Componentize the footer

### PR DESCRIPTION
**Pull request by bcomnes**
This PR is work involved to cache and componentize the footer stuff.  

I'm going to try and rope https://github.com/hypermodules/hyperamp/issues/119 and https://github.com/hypermodules/hyperamp/issues/120 into this. 

@ungoldman any thoughts on the use of cache-component and the class pattern?

### CODE REVIEW

File: renderer/elements/footer/index.js [hunk 1]
```diff
@@ -2,27 +2,15 @@ var html = require('choo/html')
...
-  max: 100,
-  step: 0.1
-}
-
-function play (state, emit) {
-  if (state.player.playing) return emit('player:pause')
-  emit('player:play')
-}
+var PlaybackCluster = require('../playback')
+var playbackCluster = new PlaybackCluster()
```
Comment (ungoldman) [inline renderer/elements/footer/index.js]: what is a playback cluster?

Comment (ungoldman) [inline renderer/elements/footer/index.js]: never mind I get it

File: renderer/elements/playback/index.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+PlaybackCluster.prototype._handleNext = function () {
+  this._emit('player:next')
+}
+
+PlaybackCluster.prototype._handlePlayPause = function () {
+  if (this._playing) this._emit('player:pause')
+  else this._emit('player:play')
+}
```
Comment (ungoldman) [inline renderer/elements/playback/index.js]: The prototype stuff seemed a little verbose but after re-reading cache-component, this makes sense. LGTM 👍

**Comment by bcomnes** [comment]
Merged with https://github.com/hypermodules/hyperamp/pull/159

**Pull request closed without merging**